source/write/gridfs.txt

```diff
@@ -0,0 +1,389 @@
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/write/gridfs.php
+      :start-after: start-retrieve-file-info
+      :end-before: end-retrieve-file-info
+      :language: php
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      {"_id":{"$oid":"..."},"chunkSize":1048576,"filename":"my_file",
+      "length":13,"uploadDate":{"$date":{"$numberLong":"..."}},"metadata":
+      {"contentType":"text\/plain"},"md5":"6b24249b03ea3dd176c5a04f037a658c"}
+
+The ``find()`` method accepts various query specifications. You can use its
+``$options`` parameter to specify the sort order, maximum number of documents to return,
+and the number of documents to skip before returning. To view a list of available
+options, see the `API documentation <{+api+}/method/MongoDBGridFSBucket-find/#parameters>`__.
+
+.. _gridfs-download-files:
+
+Download Files
+--------------
+
+You can download files from a GridFS bucket by using the following methods:
+
+- ``MongoDB\GridFS\Bucket::openDownloadStreamByName()`` or
+  ``MongoDB\GridFS\Bucket::openDownloadStream()``: Opens a new download stream
+  from which you can read the file contents
+- ``MongoDB\GridFS\Bucket::downloadToStream()``: Writes the entire file to
+  an existing download stream
+
+Read From a Download Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can download files from your MongoDB database by using the
+``MongoDB\GridFS\Bucket::openDownloadStreamByName()`` method to
+create a download stream.
+
+This example uses a download stream to perform the following actions:
+
+- Selects a GridFS file named ``'my_file'``, uploaded in the
+  :ref:`gridfs-upload-files` example, and opens it as a readable stream 
+- Calls the ``stream_get_contents()`` method to read the contents of ``'my_file'``
+- Prints the file contents
+- Calls the ``fclose()`` method to close the download stream pointing to ``'my_file'`` 
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/write/gridfs.php
+      :start-after: start-open-download-stream-name
+      :end-before: end-open-download-stream-name
+      :language: php
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      "Data to store"
+
+.. note::
+
+   If there are multiple documents with the same file name,
+   GridFS will stream the most recent file with the given name (as
+   determined by the ``uploadDate`` field).
+
+Alternatively, you can use the ``MongoDB\GridFS\Bucket::openDownloadStream()``
+method, which takes the ``_id`` field of a file as a parameter:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start-download-files-id
+   :end-before: end-download-files-id
+
+.. note::
+
+   The GridFS streaming API cannot load partial chunks. When a download
+   stream needs to pull a chunk from MongoDB, it pulls the entire chunk
+   into memory. The 255-kilobyte default chunk size is usually
+   sufficient, but you can reduce the chunk size to reduce memory
+   overhead.
```

Comment:
Is this a suitable place to talk about _increasing_ the chunk size if users will frequently work with large GridFS files?

This looks like it might have been copied from somewhere, so feel free to disregard. I'm not sure if this tutorial is the appropriate place to discuss that, but I think it's more common that users would want to increase the default chunk size than reduce it further.

Reply:
Added a note about increasing the chunk size